### PR TITLE
[docs] Add Mono Library Mode and update Mono Profiled AOT

### PR DESCRIPTION
**Pull request by mdh1418**
This PR looks to add/update documentation on [Mono's Library Mode](https://github.com/dotnet/runtime/issues/79377) and [Mono's Profiled AOT](https://github.com/dotnet/runtime/issues/69512) workflows.

### CODE REVIEW

File: docs/design/mono/mono-library-mode.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+Library Mode on Mono
+===
+
+# Background
+
+For many native applications, accessibility to bountiful APIs from .NET runtime libraries can save developers from "reinventing the wheel" in the target platform's native language. That is where [interopability](https://learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/interop/) comes in handy to access modern .NET APIs from the native side. The .NET runtime libraries require the .NET runtime to function properly, and integrating the entire .NET ecosystem may prove cumbersome and unnecessary. Instead, for a smaller footprint and more seamless experience, the runtime and custom managed code invoking .NET APIs can be bundled into a library for direct consumption. In line with [Native code interop with Native AOT](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/interop), as of .NET 8, the [mono runtime supports a library mode](https://github.com/dotnet/runtime/issues/79377) enabling mobile developers to leverage modern .NET APIs in their mobile applications with a single static or shared library.
```
Comment (lambdageek) [inline docs/design/mono/mono-library-mode.md:6]:
```suggestion
For many native applications, accessibility to bountiful APIs from .NET runtime libraries can save developers from "reinventing the wheel" in the target platform's native language. That is where [interoperability](https://learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/interop/) comes in handy to access modern .NET APIs from the native side. The .NET runtime libraries require the .NET runtime to function properly, and integrating the entire .NET ecosystem may prove cumbersome and unnecessary. Instead, for a smaller footprint and more seamless experience, the runtime and custom managed code invoking .NET APIs can be bundled into a library for direct consumption. In line with [Native code interop with Native AOT](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/interop), as of .NET 8, the [mono runtime supports a library mode](https://github.com/dotnet/runtime/issues/79377) enabling mobile developers to leverage modern .NET APIs in their mobile applications with a single static or shared library.
```

File: docs/design/mono/mono-library-mode.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+Note: The library generated from Mono's Library Mode containing custom managed code and the mono runtime will, for brevity, be referred to as the mono library.
+
+# How it works
+
+The core components of mono's library mode that enables interopability between native and managed code are as follows:
```
Comment (lambdageek) [inline docs/design/mono/mono-library-mode.md:12]:
```suggestion
The core components of mono's library mode that enables interoperability between native and managed code are as follows:
```

**Review by lambdageek — COMMENTED** [review]
Not sure if typo or if we're inventing a new term "interop-ability"

File: docs/design/mono/mono-library-mode.md (hunk 1 above)
Comment (mdh1418) [reply docs/design/mono/mono-library-mode.md:6]: Thanks! I didn't realize I've been misreading it this whole time 😅

**Review by lambdageek — APPROVED** [review]

**Pull request merged**